pkg/k8s/k8s.go

```diff
@@ -327,6 +332,75 @@ func StartBackingStore(ctx context.Context, vConfig *config.VirtualClusterConfig
 	return etcdEndpoints, etcdCertificates, nil
 }
 
+func RunCommand(ctx context.Context, command []string, component string) error {
+	writer, err := commandwriter.NewCommandWriter(component, false)
+	if err != nil {
+		return err
+	}
+	defer writer.Writer()
+
+	// start the command
+	klog.InfoS("Starting "+component, "args", strings.Join(command, " "))
+	cmd := exec.CommandContext(ctx, command[0], command[1:]...)
+	cmd.Stdout = writer.Writer()
+	cmd.Stderr = writer.Writer()
+	cmd.Cancel = func() error {
+		err := cmd.Process.Signal(os.Interrupt)
+		if err != nil {
+			return fmt.Errorf("signal %s: %w", command[0], err)
+		}
+
+		state, err := cmd.Process.Wait()
+		if err == nil && state.Pid() > 0 {
+			time.Sleep(2 * time.Second)
+		}
+
+		err = cmd.Process.Kill()
+		if err != nil {
+			return fmt.Errorf("kill %s: %w", command[0], err)
+		}
+
+		return nil
+	}
+	err = cmd.Run()
+
+	// make sure we wait for scanner to be done
+	writer.CloseAndWait(ctx, err)
+	return fmt.Errorf("error running command %s: %w", command[0], err)
+}
+
+func ExecTemplate(templateContents string, name, namespace string, values *vclusterconfig.Config) ([]byte, error) {
```

Comment:
I moved this function to the k8s package as the k0s package was removed and it's required to  sync manifest template to the virtual cluster.

https://github.com/loft-sh/vcluster/blob/main/test/e2e/manifests/init.go#L28